pcsx2/x86/microVU_Macro.inl

```diff
@@ -82,29 +96,26 @@ void endMacroOp(int mode)
 	{
 		xMOVSS(ptr32[&vu0Regs.VI[REG_Q].UL], xmmPQ);
 	}
-	if (mode & 0x10) // Status/Mac Flags were Updated
-	{
-		// Normalize
-		mVUallocSFLAGc(eax, gprF0, 0);
-		xMOV(ptr32[&vu0Regs.VI[REG_STATUS_FLAG].UL], eax);
-	}
 
 	microVU0.regAlloc->flushAll();
 	_clearNeededCOP2Regs();
 
-	if (mode & 0x10) // Update VU0 Status/Mac instances after flush to avoid corrupting anything
+	if (mode & 0x10)
 	{
-		int t0reg = _allocTempXMMreg(XMMT_INT, -1);
-		mVUallocSFLAGd(&vu0Regs.VI[REG_STATUS_FLAG].UL);
-		xMOVDZX(xRegisterSSE(t0reg), eax);
-		xSHUF.PS(xRegisterSSE(t0reg), xRegisterSSE(t0reg), 0);
-		xMOVAPS(ptr128[&microVU0.regs().micro_statusflags], xRegisterSSE(t0reg));
-
-		xMOVDZX(xRegisterSSE(t0reg), ptr32[&vu0Regs.VI[REG_MAC_FLAG].UL]);
-		xSHUF.PS(xRegisterSSE(t0reg), xRegisterSSE(t0reg), 0);
-		xMOVAPS(ptr128[&microVU0.regs().micro_macflags], xRegisterSSE(t0reg));
-		_freeXMMreg(t0reg);
+		if (!CHECK_COP2_FLAGHACK || g_pCurInstInfo->info & EEINST_COP2_NORMALIZE_STATUS_FLAG)
+		{
+			// Normalize
+			mVUallocSFLAGc(eax, gprF0, 0);
+			xMOV(ptr32[&vu0Regs.VI[REG_STATUS_FLAG].UL], eax);
+		}
+		else
+		{
+			// backup denormalized flags for the next instruction
+			// this is fine, because we'll normalize them again before this reg is accessed
+			xMOV(gprF0, ptr32[&vuRegs->VI[REG_STATUS_FLAG].UL]);
```

Comment:
Operands are backwards (this is what breaks the BIOS) but correcting it re-breaks Tekken Tag